### PR TITLE
Fix i18n: message catalogs are not compiled when specific build (refs: #4560)

### PR DESCRIPTION
**Pull request by tk0miya**
sphinx-build skips message catalog builds if specific filename are given
as arguments.

This is a fix for what I mentioned below (for 1.7-release):
https://github.com/sphinx-doc/sphinx/pull/4560#discussion_r166504204

### CODE REVIEW

**Review by stephenfin — APPROVED** [review]
Alright, this duplicates what I've done, but it's a smaller fix so let's go with this. I would have preferred if you'd asked me to backport my change instead though, given that I'd done all the work to identify this issue 😞

**Comment by tk0miya** [comment]
@stephenfin I'm very sorry. I really did not intend tohinder your work. I considered this is very small part of #4560 (As I commented yesterday, I'd considered it is refactoring PR which contains bug fix. It was my wrong.).

File: sphinx/builders/__init__.py [hunk 1]
```diff
@@ -255,11 +255,15 @@ def compile_specific_catalogs(self, specified_files):
...
-            return dom
+            docname = self.env.path2doc(path.abspath(fpath))
+            if docname:
+                return find_catalog(docname, self.config.gettext_compact)
+            else:
+                return None
 
         specified_domains = set(map(to_domain, specified_files))
+        specified_domains.discard(None)
+        print(specified_domains)
```
Comment (tk0miya) [inline sphinx/builders/__init__.py]: debug print!

**Comment by stephenfin** [comment]
> @stephenfin I'm very sorry. I really did not intend tohinder your work. I considered this is very small part of #4560 (As I commented yesterday, I'd considered it is refactoring PR which contains bug fix. It was my wrong.).

That's OK. These things happen. I'm happy with this particular change, in any case. I'll repropose the remaining parts of #4560 once this has merged and been brought back into `master`

**Comment by tk0miya** [comment]
Thanks!

**Pull request merged**